src/shared/workers/downloadHelper.ts

```diff
@@ -0,0 +1,69 @@
+/*
+    Background:
```

Comment:
For the new/old batch of reviewers:
*this PR has been open for awhile, since the backend changes took time.
* This PR is ready now, but there is a whole history of why this approach was chosen. This in-code comment describes the why.
* Additionally, spun up a minimal js fullstack app to test other approaches (where I could ctrl the full stack without any server middleware doing header stripping) -- and this was still the only approach found which worked.

Feel free to suggest/discover other approaches. Open to other options, as long as it works!